update doc references

haakon-e opened the pull request:
Add `(@ref)` to methods referenced in docstring of `methods`, `which`, and `methodswith`.

haakon-e committed:
update doc references

StefanKarpinski reviewed (approved):
LGTM, presuming it passes tests (i.e. docs actually build).

haakon-e commented:
It appears cross references fail to be discovered, [e.g.](https://buildkite.com/julialang/julia-master/builds/19850#0185892a-70d0-4ae1-bfd1-a0f78964083e/2388-2436):
```julia
┌ Error: no doc found for reference '[`@which`](@ref)' in src\base\base.md.
--
  | └ @ Documenter.CrossReferences C:\workdir\doc\deps\packages\Documenter\yf96B\src\Utilities\Utilities.jl:32
  | ┌ Error: no doc found for reference '[`@edit`](@ref)' in src\base\base.md.
  | └ @ Documenter.CrossReferences C:\workdir\doc\deps\packages\Documenter\yf96B\src\Utilities\Utilities.jl:32
  | ┌ Error: no doc found for reference '[`@which`](@ref)' in src\base\base.md.
  | └ @ Documenter.CrossReferences C:\workdir\doc\deps\packages\Documenter\yf96B\src\Utilities\Utilities.jl:32
  | ┌ Error: no doc found for reference '[`methodswith`](@ref)' in src\base\base.md.
  | └ @ Documenter.CrossReferences C:\workdir\doc\deps\packages\Documenter\yf96B\src\Utilities\Utilities.jl:32
```

perhaps more specific `@ref`s are needed ([i.e.](https://documenter.juliadocs.org/stable/man/syntax/#Named-doc-@refs)). I'll investigate.

vtjnash commented on base/reflection.jl:
```diff
@@ -1011,7 +1011,7 @@ A list of modules can also be specified as an array.
     At least Julia 1.4 is required for specifying a module.
 
-See also: [`which`](@ref) and `@which`.
+See also: [`which`](@ref), [`@which`](@ref) and [`methodswith`](@ref).
```
```suggestion
See also: [`which`](@ref), [`InteractiveUtils.@which`](@ref) and [`InteractiveUtils.methodswith`](@ref).
```

vtjnash commented on base/reflection.jl:
```diff
@@ -1550,7 +1550,7 @@ Returns the method of `f` (a `Method` object) that would be called for arguments
 If `types` is an abstract type, then the method that would be called by `invoke` is returned.
 
-See also: [`parentmodule`](@ref), and `@which` and `@edit` in [`InteractiveUtils`](@ref man-interactive-utils).
+See also: [`parentmodule`](@ref), and [`@which`](@ref) and [`@edit`](@ref) in [`InteractiveUtils`](@ref man-interactive-utils).
```
```suggestion
See also: [`parentmodule`](@ref), [`InteractiveUtils.@which`](@ref), and [`InteractiveUtils.@edit`](@ref).
```

vtjnash committed:
Apply suggestions from code review

Feel free to fix these in a different way, but I think this is one way that should resolve the reference.

haakon-e commented:
Thanks for trying to update cross-references. It doesn't seem to have fixed the issue. From [aarch-64-apple build](https://buildkite.com/julialang/julia-master/builds/20392#0185d7f2-5dbd-4d52-ad79-071c8faf42f1/1341-1380):
```julia
[ Info: SetupBuildDirectory: setting up build directory.
--
  | [ Info: Doctest: skipped.
  | [ Info: ExpandTemplates: expanding markdown templates.
  | [ Info: CrossReferences: building cross-references.
  | ┌ Error: unable to get the binding for '[`InteractiveUtils.@which`](@ref)' in src/base/base.md from expression ':(#= none:1 =# InteractiveUtils.@which)' in module Base
  | │   exception = UndefVarError: `InteractiveUtils` not defined
  | └ @ Documenter.CrossReferences /Users/administrator/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-aarch64-1.0/build/default-macmini-aarch64-1-0/julialang/julia-master/doc/deps/packages/Documenter/yf96B/src/Utilities/Utilities.jl:32
  | ┌ Error: unable to get the binding for '[`InteractiveUtils.@edit`](@ref)' in src/base/base.md from expression ':(#= none:1 =# InteractiveUtils.@edit)' in module Base
  | │   exception = UndefVarError: `InteractiveUtils` not defined
  | └ @ Documenter.CrossReferences /Users/administrator/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-aarch64-1.0/build/default-macmini-aarch64-1-0/julialang/julia-master/doc/deps/packages/Documenter/yf96B/src/Utilities/Utilities.jl:32
  | ┌ Error: unable to get the binding for '[`InteractiveUtils.@which`](@ref)' in src/base/base.md from expression ':(#= none:1 =# InteractiveUtils.@which)' in module Base
  | │   exception = UndefVarError: `InteractiveUtils` not defined
  | └ @ Documenter.CrossReferences /Users/administrator/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-aarch64-1.0/build/default-macmini-aarch64-1-0/julialang/julia-master/doc/deps/packages/Documenter/yf96B/src/Utilities/Utilities.jl:32
  | ┌ Error: unable to get the binding for '[`InteractiveUtils.methodswith`](@ref)' in src/base/base.md from expression ':(InteractiveUtils.methodswith)' in module Base
  | │   exception = UndefVarError: `InteractiveUtils` not defined
  | └ @ Documenter.CrossReferences /Users/administrator/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-aarch64-1.0/build/default-macmini-aarch64-1-0/julialang/julia-master/doc/deps/packages/Documenter/yf96B/src/Utilities/Utilities.jl:32
  | [ Info: CheckDocument: running document checks.
  | [ Info: Populate: populating indices.
  | ERROR: LoadError: `makedocs` encountered an error (:cross_references). Terminating build before rendering.
  | Stacktrace:
[...]
```

DilumAluthge commented:
Removing the `merge me` label because the CI failures are caused by the contents of the PR.

vtjnash committed:
fixup! Apply suggestions from code review

vtjnash commented:
Updated, following the example at https://documenter.juliadocs.org/stable/man/syntax/#Named-@refs, so hopefully this time works
